src/main/cc/proxy_client/proxy_client.cc

```diff
@@ -157,6 +163,43 @@ int GetInputsFromIncludeProcessor(const string& cmd_id, int argc, char** argv, c
   return 0;
 }
 
+
+string readFile(const std::string& fileName) {
+  ifstream fileStream(fileName.c_str());
+  return string((istreambuf_iterator<char>(fileStream)),
+            istreambuf_iterator<char>());
+}
+
+bool isFileArg(const std::string& str) {
+  if (str.substr(0, 5) == FILE_ARG_PREFIX) {
+    return true;
+  }
+  return false;
+}
+
+// expandFileArguments goes through the set of args given as input, and for any arg
+// that contains a "file:<absolute-file-path>" syntax, reads the file pointed to by the
+// absolute-file-path and insert the arguments specified by it into the given set of args.
+//
+// The file contents is expected to be a comma separated list of arguments. For example,
+// the following is a valid file:
+// foo/bar/a.java,bar/foo/b.h,bar/foo/b.cpp
+void expandFileArguments(set<string>* args) {
```

Comment:
Another minor google style guide quip (applies to other functions here as well): function names should be CamelCase (e.g. capitalize first letter).

Reply:
Good to know! Done.